### PR TITLE
137 strongly typed minimal apis

### PR DESCRIPTION
**Pull request by danielmackay**
﻿> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ PBI - https://github.com/SSWConsulting/SSW.CleanArchitecture/issues/137

> 2. What was changed?

✏️ 
- Switched Ardalis.Result for ErrorOr.  
- Finished migration to Result Pattern.  
- Removed Strongly Typed APIs (mostly).

> 3. Did you do pair or mob programming?

✏️ No

### CODE REVIEW

File: src/WebApi/Extensions/EndpointRouteBuilderExt.cs [hunk 1]
```diff
@@ -35,4 +34,10 @@ public static RouteHandlerBuilder ProducesDelete(this RouteHandlerBuilder builde
         .Produces(StatusCodes.Status204NoContent)
         .Produces(StatusCodes.Status404NotFound)
         .ProducesProblem(StatusCodes.Status500InternalServerError);
+
+    /// <summary>
+    /// TODO: Remove once we have a global way to configure Open API with a 500 response for all endpoints
```
Comment (matt-goldman-ssw) [inline src/WebApi/Extensions/EndpointRouteBuilderExt.cs:39]: Is this not a global way to do this?

File: src/WebApi/Extensions/ErrorOrExt.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+using ErrorOr;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace SSW.CleanArchitecture.WebApi.Extensions;
+
+public static class ErrorOrExt
```
Comment (matt-goldman-ssw) [inline src/WebApi/Extensions/ErrorOrExt.cs:6]:
```suggestion
public static class ErrorOrExtensions
```

Prefer something a bit more verbose, we don't have a character restriction here

**Review by matt-goldman-ssw — APPROVED** [review]
LGTM - left some comments for your consideration

File: src/WebApi/Extensions/ErrorOrExt.cs (hunk 2 above)
Comment (danielmackay) [reply src/WebApi/Extensions/ErrorOrExt.cs]: I ended up renaming to `CustomResult`.  Originally this was an extension method, but is not anymore.

**Pull request merged**